### ISSUE TITLE
RBAC role to provide project-owner permissions while limiting quota management

### ISSUE THREAD

**Issue by gaktive** [rancher/dashboard#5417]
Internal reference: SURE-4088
Reported in 2.6.2

Issue description:

A user would like to continue using a custom project role (closely aligned with project-owner) used in Rancher v2.5.x, upgrade tests indicate the custom role doesn't allow the ability to manage the project.

The role is designed to:
- Allow members with the role to manage other members (as with project-owner)
- Prevent those same members from managing project quotas

Business impact:
Prevents upgrade to v2.6.x

Troubleshooting steps:
Tried different permissions changes, notably get, list, watch verbs for `projects.management.cattle.io` (bellow).

``` 
- apiGroups:
  - management.cattle.io
  resources:
  - projects
  verbs:
  - get
  - list
  - watch
```

Without write permissions (update, patch, etc.) the Rancher UI only shows "View Config" when trying to edit the project. This is perhaps expected, however, the user does have write permissions for `projectroletemplatebindings` to manage users, which is also managed in the same area of the UI.

Repro steps:

- Create a local user with Standard global permissions
- Create or edit an existing project to add the above user as a Project Admin (if using the same naming as the attached example custom project role)
- Login as the user, navigate to the project and attempt to edit

Workaround:
Is workaround available and implemented? no

Unlikely, unless a relaxing of permissions is acceptable.

Actual behavior:

Without write permissions for projects, the custom project role is unable to manage project membership (more correctly, the "Edit Config" options is unavailable on the project).

Expected behavior:
The custom project role can use "Edit Config" to edit project membership, changes to project quotas will be denied.

Additional notes:
The user reported that the 'Create Namespace' button wasn't available at times when using the custom role. This was also noticed this in local testing. This may be a caching issue, or if a namespace did not already exist in the project

**Comment by aalves08** [comment]
@gaktive upon further test while reviewing we have found, imo, a blocking issue where the user with custom role the customer provided (**Project Admin**) are able to elevate themselves to Project Owner.

Still trying to replicate it (I was on a call with Richard when he reproduced it consistently) but the BE allowed for the operation which shouldn't happen.

With that in mind, we might need assistance from someone on the BE side to try to assist us.

There's also another issue, this one is FE related, where we have to disable the edit button per project (and not globally like I was doing) for projects in which that user isn't allowed to enter. This is a tricky subject and requires some time to tackle it.

These issues only came up on this final round of testing with much more complex scenarios of roles/resource assignments to that **Project Admin** user.

I'll try to add the exact conditions we found the blocking issue, as soon as I have figured it out, but clearly this RBAC is at risk for 2.6.5 🙏 

FYI: @richard-cox @kwwii

**Comment by gaktive** [comment]
@aalves08 ok, once you do replicate it, make sure you file a ticket in `rancher/rancher` with what you need help with on the backend and link back here for context. A user being able to elevate permissions shouldn't be happening, so that's a good thing to discover.

**Comment by richard-cox** [comment]
Fresh v2.6-head (68eb5d1)

**Scenario 1**

As Admin

1. Import `project admin` role from https://github.com/rancher/dashboard/pull/5664
   - This is the example used by customer who had previously seen this working(?) 
2. Create `user1` as `user-base`
3. Create `project1`, at create time add `user-base` as a `project-admin`

As `user1` and running dashboard with #5664

4. Attempt to edit `project1`. User cannot change roles as they cannot GET `management.cattle.io.roletemplate`. 

**Scenario 2**
Following on from scenario 1

As Admin

1. Create `user2` as `standard-user`
2. Edit `project1` and assign `user2` as `project admin`

As `user2` and running dashboard with #5664

3. Edit `project1` and assign `user2` (themselves) as `Project Owner`. This succeeds (which defeats the point of the request in the issue)

From our side, we need to understand if there's a set of resource permissions that can be wrapped in a role that would block them from editing project specific settings but allow them to manage other users within the project... without exceeding their own privileges. This might be possible with the provided role and there's a backend bug or this might mean the role needs to be changed to achieve the task. At this point we need BE assistance.

**Comment by aalves08** [comment]
@gaktive after a sync with @nwmac and @richard-cox we decided to push forward the work already done to 2.6.5, mostly because the old ember UI already allowed for this particular edge-case.

A couple of follow-up issues will arise (_we will link it to this for context_) which will be dealt with for 2.6.6, but this should unblock the user to upgrade to 2.6-head. 🙏 

Thanks

FYI: @kwwii

**Comment by sowmyav27** [comment]
**On 2.6.4**

1. create two users - both standard users, one to be project admin (user1) and another just for testing (user2
2. create a new role as Project Admin, as mentioned [here](https://github.com/rancher/dashboard/pull/5664#issue-1209500917)
3. As admin, create a new project, then assign the user user1 to the project with role Project Admin
4. In an incognito browser window, login as user user1 and edit the project which was assigned to the user
5. Edit Config is NOT available as an option for User1

---

**On 2.5.13**

1. create two users - both standard users, one to be project admin (user1) and another just for testing (user2
2. create a new role as Project Admin, as mentioned [here](https://github.com/rancher/dashboard/pull/5664#issue-1209500917)
3. As admin, create a new project, then assign the user user1 to the project with role Project Admin
4. In an incognito browser window, login as user user1 and edit the project which was assigned to the user
5. Edit the members list by adding test user with custom project permissions -> select only View Project Catalogs (the user giving access may only provide credentials for which he has access to in terms of roles/actions)
6. after hitting save, user2 is added to the project by user1 (ember UI)

---

**On 2.6-head commit id: `311f462`**
1. create two users - both standard users, one to be project admin (user1) and another just for testing (user2
2. create a new role as Project Admin, as mentioned [here](https://github.com/rancher/dashboard/pull/5664#issue-1209500917)
3. As admin, create a new project, then assign the user user1 to the project with role Project Admin
4. In an incognito browser window, login as user user1 and edit the project which was assigned to the user
5. Edit the members list by adding test user with custom project permissions -> select only View Project Catalogs (the user giving access may only provide credentials for which he has access to in terms of roles/actions)
6. after hitting save, user2 is added to the project by user1 (ember UI)

**On an Upgrade from 2.5.13 to 2.6-head commit id `311f462`**
1. create two users - both standard users, one to be project admin (user1) and another just for testing (user2
2. create a new role as Project Admin, as mentioned [here](https://github.com/rancher/dashboard/pull/5664#issue-1209500917)
3. As admin, create a new project, then assign the user user1 to the project with role Project Admin
4. In an incognito browser window, login as user user1 and edit the project which was assigned to the user
5. Edit the members list by adding test user with custom project permissions -> select only View Project Catalogs (the user giving access may only provide credentials for which he has access to in terms of roles/actions)
6. after hitting save, user2 is added to the project by user1 (ember UI)
7. Upgrade rancher to 2.6head
8. User 1 continues to have permissions to Edit config of the project
9. As Admin create another user - user3
10. Login as user3. Edit the members list by adding test user with custom project permissions -> select only View Project Catalogs (the user giving access may only provide credentials for which he has access to in terms of roles/actions)
11.  after hitting save, user3 is added to the project by user1 (ember UI)

---

**On upgrade from 2.6.4 to 2.6-head commit id `311f462`**
1. create two users - both standard users, one to be project admin (user1) and another just for testing (user2
2. create a new role as Project Admin, as mentioned [here](https://github.com/rancher/dashboard/pull/5664#issue-1209500917)
3. As admin, create a new project, then assign the user user1 to the project with role Project Admin
4. In an incognito browser window, login as user user1 and edit the project which was assigned to the user
5. Edit Config is NOT available as an option for User1
6. Upgrade rancher to 2.6head
8. User 1 has permissions to Edit config of the project
9. Edit the members list by adding test user with custom project permissions -> select only View Project Catalogs (the user giving access may only provide credentials for which he has access to in terms of roles/actions)
10. after hitting save, user1 is added to the project by user1 (ember UI)
